src/block_disagg/block_disagg_write.c

```diff
@@ -110,7 +110,14 @@ __wt_block_disagg_write_internal(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *bloc
     page_id = block_meta->page_id;
 
     /* Get the checkpoint ID. */
-    checkpoint_id = 1; /* XXX Use the current checkpoint order instead of hardcoding this. */
+    checkpoint_id = S2C(session)->disaggregated_storage.global_checkpoint_id;
```

Comment:
Should this come from the block meta? If not, does this mean the checkpoint id in block meta is only for read? Should we update it after the write?

Reply:
No, the `block_meta` is the metadata from the previous version of the page.

Reply:
This is different to my implementation. I pass the block meta as the new block meta. For example, the delta count is increased before I pass the block meta in. After the reconciliation, the block meta is assigned to the page's block meta as the current page's block meta. If we don't pass in a new block meta, how do we know we need to write a delta?

Reply:
This method was intended to take previous metadata in `block_meta` and puts new metadata into `new_block_meta`; you can see it populated at the end of the function. We should pass delta differently?

Reply:
We need to tell the function that we want to write a delta. That is currently done by passing the block_meta. For a new page that hasn't written before, I also pass in the block_meta to tell the new page id.

Reply:
The reason for this design was that it would simplify managing the block metadata by doing all of it just in one place.

Reply:
As discussed on Zoom, we'll fix this in a separate ticket. The conclusion was to get the btree code to own changing the page block meta struct, perhaps with the only exception being the LSN can be filled out by the block manager. Thanks Chenhao!